### PR TITLE
Easier bulk editing and code simplification

### PR DESCRIPTION
**Pull request by sleidig**

### CODE REVIEW

File: src/couchdb/couchdb-admin.controller.ts [hunk 1]
```diff
@@ -1,147 +1,135 @@
...
+        const configString = JSON.stringify(config);
+        const regex = new RegExp(searchString, 'g');
 
-      if (configString.match(regex)) {
-        editedOrgs.push(cred.name);
-      }
-    }
-    return editedOrgs;
+        if (configString.match(regex)) {
+          return 'edited';
```
Comment (TheSlimvReal) [inline src/couchdb/couchdb-admin.controller.ts:53]:
```suggestion
          return 'matching';
```

File: src/couchdb/couchdb.service.ts [hunk 2]
```diff
@@ -12,48 +15,91 @@ export class CouchdbService {
...
+  }
+
+  /**
+   * Run a function for all given databases.
+   * @param credentials
+   * @param callback A function that is called repeatedly, receiving the Couchdb instance for each database;
+   *                  return values are mapped into a key-value return object (key = credentials.name).
+   */
+  async runForAllOrgs(
+    credentials: { name: string; password: string }[],
```
Comment (TheSlimvReal) [inline src/couchdb/couchdb.service.ts:29]: Is there a benefit to providing the credentials as parameter? All the calls provide the same credentials anyways.

File: src/couchdb/migration.controller.ts [hunk 3]
```diff
@@ -1,81 +1,69 @@
...
     const logo = await fetch(url).then((res) => res.blob());
     const logoBuffer = await logo.arrayBuffer();
     const resizedLogo = await sharp(logoBuffer)
       .resize({ width: 300 })
       .toBuffer();
-    const resizedBlob = new Blob([new Uint8Array(resizedLogo)], {
+    const resizedBlob = new Blob([new Uint8Array(resizedLogo as any)], {
       type: logo.type,
     });
     const attachmentsPath = '/app-attachments/SiteSettings:general';
```
Comment (TheSlimvReal) [inline src/couchdb/migration.controller.ts:64]:
```suggestion
    const attachmentsPath = '/app-attachments/SiteSettings:global';
```
Good find that this was incorrect.

**Review by TheSlimvReal — APPROVED** [review]
Really nice refactoring. This will make it much easier to write server-side migrations in the future!

File: src/couchdb/couchdb.service.ts (hunk 2 above)
Comment (sleidig) [reply src/couchdb/couchdb.service.ts:29]: I was thinking to add a query param to controllers in the future in order to let us filter which org it's applied to. (I was struggling to run migration for a single instance for testing.) If that's done in the controller, I think it'll be easier

**Pull request merged**